src/ai/backend/manager/api/vfolder.py

```diff
@@ -82,6 +85,63 @@
 VFolderRow = Mapping[str, Any]
 
 
+def vfolder_filter_by_status_required(perm: VFolderOperationStatus):
+    """
+    Checks if the target vfolder status is READY.
+    The decorated handler should prevent user access
+    while storage-proxy operations such as vfolder clone, deletion is handling.
+    """
+
+    def _wrapper(handler: Callable[..., Awaitable[web.Response]]):
+        @functools.wraps(handler)
+        async def _wrapped(request: web.Request, *args, **kwargs) -> web.Response:
+            root_ctx: RootContext = request.app["_root.context"]
+            domain_name = request["user"]["domain_name"]
+            user_role = request["user"]["role"]
+            user_uuid = request["user"]["uuid"]
+            allowed_vfolder_types = await root_ctx.shared_config.get_vfolder_types()
+            folder_name = request.match_info["name"]
+            vf_name_conds = vfolders.c.name == folder_name
+            if perm == VFolderOperationStatus.READY:
```

Comment:
Please see comment below.

Reply:
@kyujin-cho Thanks for the review! I've updated commits below. 
I have an additional question about using this vfolder filter wrapper method. 
Would it be necessary to add the decorator to all of the operations defined in vfolder api?
(commits above declare decorator only for "ready to alter" related operations but not for "ready to read" operations like `get_*`)

Reply:
Finding simillar definitions on previous approaches can be a good starting point. In Backend.AI manager, we have a decorator function `server_status_required()`, which limits access from user to API by filtering user’s permission. When you look at the usages of `server_status_required()`, you can see that we have added this to all of the API handlers which requires authentication, even though it’s allowed to all users, not only for previliged users (admin or superadmin). I think this will be a good reference for `vfolder_filter_by_status_required()`.      
So, in short: yes, please add it to all vFolder API handlers!

Reply:
updated this in c14de5e!